testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -645,4 +646,112 @@ public void testReadSparkBucketedTable()
+                    ") partitioned by (p_bigint, p_varchar)", tableName));
+
+            onSpark().executeQuery("set hive.exec.dynamic.partition.mode=nonstrict");
+            onSpark().executeQuery(format("INSERT INTO %s VALUES " +
+                    "(120, 32760, 2147483640, 9223372036854775800, 123.340, 234.560, CAST(343.0 AS DECIMAL(10, 0)), CAST(345.670 AS DECIMAL(10, 5)), TIMESTAMP '2015-05-10 12:15:30', DATE '2015-05-08', 'p1 varchar', CAST('varchar10' AS VARCHAR(10)), CAST('p1 char10' AS CHAR(10)), false, CAST('p1 binary' as BINARY), 1, 'partition 1')," +
+                    "(120, 32760, 2147483640, 9223372036854775800, 123.340, 234.560, CAST(343.0 AS DECIMAL(10, 0)), CAST(345.670 AS DECIMAL(10, 5)), TIMESTAMP '2015-05-10 12:15:30', DATE '2015-05-08', 'p1 varchar', CAST('varchar10' AS VARCHAR(10)), CAST('p1 char10' AS CHAR(10)), false, CAST('p1 binary' as BINARY), 2, 'partition 2')," +
+                    "(null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 2, 'partition 2')," +
+                    "(null, null, null, null, null, null, null, null, null, null, null, null, null, null, null, 1, 'partition 1')", tableName));
+            onSpark().executeQuery(format("ANALYZE TABLE %s COMPUTE STATISTICS FOR ALL COLUMNS", tableName));
+            onSpark().executeQuery(format("ANALYZE TABLE %s PARTITION(p_bigint, p_varchar) COMPUTE STATISTICS", tableName));
+            assertThat(onTrino().executeQuery(format("SHOW STATS FOR %s", tableName))).containsOnly(ImmutableList.of(
+                    row("c_tinyint", null, 1.0, 0.5, null, "120", "120"),
+                    row("c_smallint", null, 1.0, 0.5, null, "32760", "32760"),
+                    row("c_int", null, 1.0, 0.5, null, "2147483640", "2147483640"),
+                    row("c_bigint", null, 1.0, 0.5, null, "9223372036854775807", "9223372036854775807"),
+                    row("c_float", null, 1.0, 0.5, null, "123.34", "123.34"),
+                    row("c_double", null, 1.0, 0.5, null, "234.56", "234.56"),
+                    row("c_decimal", null, 1.0, 0.5, null, "343.0", "343.0"),
+                    row("c_decimal_w_params", null, 1.0, 0.5, null, "345.67", "345.67"),
+                    row("c_timestamp", null, 1.0, 0.5, null, null, null),
+                    row("c_date", null, 1.0, 0.5, null, "2015-05-08", "2015-05-08"),
+                    row("c_string", 20.0, 1.0, 0.5, null, null, null),
+                    row("c_varchar", 18.0, 1.0, 0.5, null, null, null),
+                    row("c_char", 20.0, 1.0, 0.5, null, null, null),
+                    row("c_boolean", null, 1.0, 0.5, null, null, null),
+                    row("c_binary", 18.0, null, 0.5, null, null, null),
+                    row("p_bigint", null, 2.0, 0.0, null, "1", "2"),
+                    row("p_varchar", 44.0, 2.0, 0.0, null, null, null),
+                    row(null, null, null, null, 4.0, null, null)));
```

Comment:
data_size column shouldn't remain null

Reply:
Right now, `data_size` is always set to null. This should be changed in separated PR.